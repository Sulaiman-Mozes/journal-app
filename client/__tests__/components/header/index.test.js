import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Header from '../../../components/header';

describe('Header Component', () => {
  const props = {
    isloggedIn: true,
    logout: jest.fn(),
    username: 'sulaiman_mozes',
  };
  const header = shallow(<Header {...props} />);
  const headerNotLoggedIn = shallow(<Header {...props} isloggedIn={false} />);

  it('renders <Header/> correctly', () => {
    expect(header).toMatchSnapshot();
    expect(headerNotLoggedIn).toMatchSnapshot();
  });
});
