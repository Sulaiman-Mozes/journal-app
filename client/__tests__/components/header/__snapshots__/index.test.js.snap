// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders <Header/> correctly 1`] = `
<header>
  <nav
    className="navbar  fixed-top navbar-expand-lg navbar-dark py-4 indigo"
  >
    <NavLink
      className="navbar-brand"
      to="/dashboard"
    >
      My Journal
    </NavLink>
    <button
      aria-controls="navbarText"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarText"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarText"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeStyle={
              Object {
                "color": "#F15B2A",
              }
            }
            className="nav-link"
            to="/about"
          >
            About
          </NavLink>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <div
            className="nav-link"
          >
            Welcome Â 
            sulaiman_mozes
          </div>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeStyle={
              Object {
                "color": "#F15B2A",
              }
            }
            className="nav-link"
            onClick={[MockFunction]}
            to="/login"
          >
            Logout
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;

exports[`Header Component renders <Header/> correctly 2`] = `
<header>
  <nav
    className="navbar  fixed-top navbar-expand-lg navbar-dark py-4 indigo"
  >
    <NavLink
      className="navbar-brand"
      to="/"
    >
      My Journal
    </NavLink>
    <button
      aria-controls="navbarText"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarText"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarText"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeStyle={
              Object {
                "color": "#F15B2A",
              }
            }
            className="nav-link"
            to="/about"
          >
            About
          </NavLink>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeStyle={
              Object {
                "color": "#F15B2A",
              }
            }
            className="nav-link"
            to="/login"
          >
            Login
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeStyle={
              Object {
                "color": "#F15B2A",
              }
            }
            className="nav-link"
            to="/signup"
          >
            Signup
          </NavLink>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
